workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/selection/VaultWatcherService.kt

```diff
@@ -38,14 +47,10 @@ class VaultWatcherService(tokenObserver: TokenObserver? = null) : SingletonSeria
 
         private fun getObservableFromAppServiceHub(appServiceHub: AppServiceHub): TokenObserver {
             val config = appServiceHub.cordappProvider.getAppContext().config
+            val configOptions = InMemorySelectionConfig.parse(config)
+            // TODO cacheSize = configOptions.cacheSize
 
-            val indexingType = try {
-                IndexingType.valueOf(config.get("ownerIndexingStrategy").toString())
-            } catch (e: Exception) {
-                IndexingType.PUBLIC_KEY
-            }
-
-            val ownerProvider: ((StateAndRef<FungibleToken>, AppServiceHub) -> Any) = if (indexingType == IndexingType.PUBLIC_KEY) {
+            val ownerProvider: ((StateAndRef<FungibleToken>, AppServiceHub) -> Any) = if (configOptions.indexingStrategy == IndexingType.PUBLIC_KEY) {
```

Comment:
We should not bucket by public key. Instead we should bucket by account OR account AND type